### PR TITLE
update to SourceLink 1.0

### PR DESCRIPTION
**Pull request by ctaggart**
I'm working on SourceLink 0.6. @forki logged https://github.com/ctaggart/SourceLink/issues/45 that I'm going to try to address here. This isn't to be merged until I release 0.6. Ideally this coming week.

### CODE REVIEW

File: paket.lock [hunk 1]
```diff
@@ -65,12 +65,11 @@ NUGET
       Nancy (>= 1.2.0)
     Newtonsoft.Json (6.0.8)
     NuGet.CommandLine (2.8.5)
-    Nuget.Core (2.8.5)
+    NuGet.Core (2.8.5)
```
Comment (ctaggart) [inline paket.lock]: It this really causing a Travis build failure. Yikes. That is the real name of the package and is what Paket-Install 1.18.0 set it to.

**Comment by ctaggart** [comment]
image

:-)

File: build.fsx [hunk 2]
```diff
@@ -143,21 +143,11 @@ Target "Test" (fun _ ->
...
-                -- "**/AssemblyInfo.fs"
-        try
-            repo.VerifyChecksums files
-            proj.VerifyPdbChecksums files
-        with
-        | _ -> ()
-        proj.CreateSrcSrv (sprintf "%s/%s/{0}/%%var2%%" gitRaw projectName) repo.Commit (repo.Paths files)
-        Pdbstr.exec proj.OutputFilePdb proj.OutputFilePdbSrcSrv
+        let url = sprintf "%s/%s/{0}/%%var2%%" gitRaw projectName
+        proj.SourceIndex proj.OutputFilePdb proj.CompilesNotLinked __SOURCE_DIRECTORY__ url
```
Comment (ctaggart) [inline build.fsx]: I think this will be a lot more straightforward for people.

File: build.fsx [hunk 3]
```diff
@@ -263,11 +253,12 @@ Target "Default" DoNothing
 "Clean"
     ==> "SetAssemblyInfo"
     ==> "BuildSolution"
-    ==> "Test"    
+    ==> "Test"
+    =?> ("SourceLink", not isLinux)
```
Comment (ctaggart) [inline build.fsx]: I'll put this back before it is ready to merge.

File: build.fsx [hunk 4]
```diff
@@ -143,22 +143,11 @@ Target "Test" (fun _ ->
...
-        try
-            repo.VerifyChecksums files
-            proj.VerifyPdbChecksums files
-        with
-        | _ -> ()
-        proj.CreateSrcSrv (sprintf "%s/%s/{0}/%%var2%%" gitRaw projectName) repo.Commit (repo.Paths files)
-        Pdbstr.exec proj.OutputFilePdb proj.OutputFilePdbSrcSrv
-    )
+        let url = sprintf "%s/%s/{0}/%%var2%%" gitRaw projectName
+        proj.SourceIndex proj.CompilesNotLinked proj.OutputFilePdb __SOURCE_DIRECTORY__ url )
```
Comment (ctaggart) [inline build.fsx]: I think this will be a lot easier for people.

File: build.fsx [hunk 5]
```diff
@@ -143,22 +143,11 @@ Target "Test" (fun _ ->
...
-        try
-            repo.VerifyChecksums files
-            proj.VerifyPdbChecksums files
-        with
-        | _ -> ()
-        proj.CreateSrcSrv (sprintf "%s/%s/{0}/%%var2%%" gitRaw projectName) repo.Commit (repo.Paths files)
-        Pdbstr.exec proj.OutputFilePdb proj.OutputFilePdbSrcSrv
-    )
+        let url = sprintf "%s/%s/{0}/%%var2%%" gitRaw projectName
+        SourceLink.Index proj.CompilesNotLinked proj.OutputFilePdb __SOURCE_DIRECTORY__ url )
```
Comment (ctaggart) [inline build.fsx]: This feels much nicer!

**Comment by forki** [comment]
is it ready to merge?

**Comment by ctaggart** [comment]
Ready to merge now.

**Pull request merged**